packages/hooks/src/useRequestPhoneNumberOTP.tsx

```diff
@@ -82,48 +81,19 @@ const useRequestPhoneNumberOTP = () => {
         };
     };
 
-    const phone_number_taken_message = (
-        <Localize
-            i18n_default_text='Number already exists in our system. Enter a new one or contact us via <0>live chat</0> for help.'
-            components={[
-                <span
-                    key={0}
-                    className='phone-verification__card--inputfield__livechat'
-                    onClick={() => window.LC_API.open_chat_window()}
-                />,
-            ]}
-        />
-    );
-    const request_other_carrier_message = (
-        <Localize
-            i18n_default_text="We're unable to send codes via {{ current_carrier }} right now. Get your code by {{other_carriers}}."
-            values={{
-                current_carrier: getCurrentCarrier(),
-                other_carriers: getOtherCarrier(),
-            }}
-        />
-    );
-    const invalid_phone_message = (
-        <Localize i18n_default_text='Enter a valid phone number, including the country code (e.g. +15417541234).' />
-    );
-
     const formatError = ({ code, message }: TFormatError) => {
-        setIsDisabledRequestButton(true);
-        switch (code) {
-            case 'PhoneNumberTaken':
-                setErrorMessage(phone_number_taken_message);
-                break;
-            case 'PhoneNumberVerificationSuspended':
-                setIsDisabledRequestButton(false);
-                setErrorMessage(request_other_carrier_message);
-                break;
-            case 'InvalidPhone':
-                setIsDisabledRequestButton(false);
-                setErrorMessage(invalid_phone_message);
-                break;
-            default:
-                setErrorMessage(message);
-                break;
+        const errorCases = {
+            PhoneNumberTaken: true,
+            PhoneNumberVerificationSuspended: false,
+            InvalidPhone: false,
```

Comment:
What do 'true' and 'false' mean in this object?

Reply:
this `true` and `false` value is for the use of `setIsDisabledRequestButton` in the hook. So for `PhoneNumberTaken` we won't have any timer to stop user on clicking the `Send code via SMS/WhatsApp` button. We will enable back the button only when user edits the field. As for `PhoneNumberVerificationSuspended` and `InvalidPhone` we will let user request again once the timer finishes running.

Reply:
can we add a comment in the code. Rest is lgtm